frontend/hooks/useNeedsFunds.ts

```diff
@@ -46,37 +47,35 @@ export const useNeedsFunds = (serviceConfigId?: string) => {
 
         const templateStakingProgramId =
           serviceTemplate.configurations[middlewareChain].staking_program_id;
-        const serviceStakingProgramId =
-          service?.chain_configs[middlewareChain as MiddlewareChain]?.chain_data
-            ?.user_params?.staking_program_id;
-        const stakingProgramId =
-          serviceStakingProgramId ?? templateStakingProgramId;
 
-        if (!stakingProgramId) return;
-        if (!service?.chain_configs[middlewareChain as MiddlewareChain]) return;
+        // if stakingProgramId not provided, use the one from the template
+        const resolvedStakingProgramId =
+          stakingProgramId ?? templateStakingProgramId;
+
+        if (!resolvedStakingProgramId) return;
```

Comment:
we should use defaultStakingProgramId here instead, template elements are generally overwritten, have put default staking program ids in a config

Reply:
good catch, used the selected one here, should work